### ISSUE TITLE
Initial ajax call to fill the table

### ISSUE THREAD

**Issue by cadilhac** [lorenzofox3/Smart-Table#242]
I added the st-pipe attribute to my table, referencing my getData function. It is never called. Looking at your code, it seems that it is called only when sorting or searching. Since this is the function that fills the grid given the tableparams, why is it not called initially?

Thanks

**Comment by cyberwombat** [comment]
Why is this closed? This sounds like pretty blatant issues. I've been looking for an alternative just because of this.

**Comment by lorenzofox3** [comment]
Because, I still don't understand the issue and the author failed to provide a way to reproduce the issue and showed he won't make any effort to help anyway (https://github.com/lorenzofox3/Smart-Table/issues/244). You can have a look at the "pipe/ajax plugin" section on the documentation as a started to reproduce the issue. But on this example it seems to work as expected, 
regards,

Laurent R

**Comment by cyberwombat** [comment]
I see. I am not sure about his second issues as I kinda stopped using this at the first one. I do find it odd behavior that the tables will not display anything unless sort/search is called. I eventually figured out I had to make a separate call by looking at the source as well. (might add to docs - I never did figure out why your demo seems to show data intially while my copy and pasting of it failed to do so). But I understand it's a design preference thing. But +1 for having the tables handle initial data load in case you are thinking about it. Thanks.

**Comment by mehulcrest** [comment]
Hi,
I am facing issue like the same.
I have code what you have explained on http://lorenzofox3.github.io/smart-table-website/#section-pipe but initial call is not working and no data coming to screen.

Static data and pagination is working which explained on http://lorenzofox3.github.io/smart-table-website/#section-pagination

Please advise.

**Comment by lorenzofox3** [comment]
ok with bb19a5778c2f451787dbd85533dd6b91976b7d55, the first call to pipe should be handled by the table. Note as mentioned @cadilhac if you were using the pagination directive, the table was kicked by itself and you should not had any problem. Now to prevent the pipe function to be called twice you'll have to set the initial pagination state in you pipe function:

```Javascript

function customPipe(tableState, ctrl){

//if you use the pagination directive make sure to set the initial pagination state to prevent a second initial //call from the pagination directive
if(!tableState.pagination.number){
tableState.pagination.number = 10;
tableState.pagination.start = 0;
}

//custom pipe logic

}
```

A bit hackish but I have not found better so far

**Comment by mehulcrest** [comment]
Hi,

My issue was that the service is not being called in this case.

functionality is working fine when we did some changes in view (html) file.

I have added one outer div with alias "ctrl" in ng-controller attribute like 
<b>ng-controller="pipeCtrl as ctrl"</b>
and 
I have replace <b>"mc"</b> with <b>"ctrl"</b>

<pre>
<b>&lt;div ng-controller="pipeCtrl as ctrl"&gt;</b>

&lt;table class="table" st-pipe="<b>ctrl</b>.callServer" st-table="<b>ctrl</b>.displayed"&gt;
            &lt;thead&gt;
            &lt;tr&gt;
                &lt;th st-sort="id"&gt;id&lt;/th&gt;
                &lt;th st-sort="name"&gt;name&lt;/th&gt;
                &lt;th st-sort="age"&gt;age&lt;/th&gt;
                &lt;th st-sort="saved"&gt;saved people&lt;/th&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
                &lt;th&gt;&lt;input st-search="'id'"/&gt;&lt;/th&gt;
                &lt;th&gt;&lt;input st-search="'name'"/&gt;&lt;/th&gt;
                &lt;th&gt;&lt;input st-search="'age'"/&gt;&lt;/th&gt;
                &lt;th&gt;&lt;input st-search="'saved'"/&gt;&lt;/th&gt;
            &lt;/tr&gt;
            &lt;/thead&gt;
            &lt;tbody ng-show="!<b>ctrl</b>.isLoading"&gt;
            &lt;tr ng-repeat="row in <b>ctrl</b>.displayed"&gt;
                &lt;td&gt;{{row.id}}&lt;/td&gt;
                &lt;td&gt;{{row.name}}&lt;/td&gt;
                &lt;td&gt;{{row.age}}&lt;/td&gt;
                &lt;td&gt;{{row.saved}}&lt;/td&gt;
            &lt;/tr&gt;
            &lt;/tbody&gt;
            &lt;tbody ng-show="<b>ctrl</b>.isLoading"&gt;
            &lt;tr&gt;
                &lt;td colspan="4" class="text-center"&gt;Loading ... &lt;/td&gt;
            &lt;/tr&gt;
            &lt;/tbody&gt;
            &lt;tfoot&gt;
            &lt;tr&gt;
                &lt;td class="text-center" st-pagination="" st-items-by-page="10" colspan="4"&gt;
                &lt;/td&gt;
            &lt;/tr&gt;
            &lt;/tfoot&gt;
        &lt;/table&gt;

<b>&lt;/div&gt;</b>
</pre>

**Comment by bassemBMI** [comment]
Hi,
To force the first load I used data-ng-init="ctrl.callServer(ctrl.tableState)" , it's work and display first 10 records but not the pagination !!
Note that I passed "ctrl.tableState" to callServer function to avoid undefined tableState error

**Comment by bassemBMI** [comment]
Finally I fixed it, I'm using requireJS and I defined datatable files inside the same controller, so my controller was loaded before datatable, after moving those files to be load with the application it works fine.

**Comment by arulmozhimanikandan** [comment]
I have a diffrent problem, in my situation i load the table on page load itself next will have post request for different functionality, this ajax post request reseting the table data. any one facing this sort of issue

**Comment by crajkumarbe** [comment]
Hi mehulcrest/lorenzofox3,

Can you please upload your working copy of "pipectrl" from the server? Share us the code in plnker or jsfiddle or upload the zip file and provide the URL to download? 

I'm facing similar kind of problem, but unable to figure out what is breaking.

**Comment by jsgoupil** [comment]
@lorenzofox3 Hey, thanks for clarifying why this is getting two requests. I have been looking forever.

However, this is way too hackish to be mainstream with your awesome smart-table. People will hit this problem and will end up on this github issue. I had to read the entire thread to finally get that I had to have 2 custom lines. This should definitely be handled differently. Your code is also incorrect, the variable is wrong.

This issue should be mentioned on the main website documentation. It is so hackish that I do feel it will pollute your documentation though.
